assets/app/components/site/editor/editorContainer.js

```diff
@@ -106,7 +133,63 @@ class Editor extends React.Component {
       path = `${path}.md`;
     }
 
-    siteActions.createCommit(site, path, content, message, sha);
+    return { content, path, message };
+  }
+
+  commitOrPR() {
+    const { site } = this.props;
+    const { path } = this.state;
+    const draftBranch = getDraft(path, site.branches);
+
+    if (!draftBranch) {
+      this.submitFile();
+    }
+
+    this.submitFile(draftBranch.name).then(() => {
+      return siteActions.createPR(site, draftBranch.name, site.defaultBranch);
+    }).then(() => {
+      routeActions.redirect(`/sites/${site.id}`);
+    });
+  }
+
+  submitFile(branch = this.props.site.defaultBranch) {
+    const { site } = this.props;
+    const { sha } = this.state;
+    const {content, path, message} = this.buildFile();
+    const nextSite = Object.assign({}, site, {
+      branch
+    });
+
+    return siteActions.createCommit(nextSite, path, content, message, sha);
+  }
+
+  submitDraft() {
+    const { site } = this.props;
+    const { path } = this.state;
+    const draftBranch = getDraft(path, site.branches);
+
+    if (draftBranch) {
+      this.submitFile(draftBranch.name);
+    } else {
+      siteActions.createDraftBranch(site, path).then((branchName) => {
+        return this.submitFile(branchName);
+      }).then(() => {
+        const fileName = path.split('/').pop();
+        const branch = formatDraftBranchName(path);
+
+        routeActions.redirect(`/sites/${site.id}/edit/${branch}/${fileName}`);
```

Comment:
We're encoding these paths multiple times in this file and the paths drive functionality. Can we extract lines like these to something that says "redirectToFileOnBranch" or a better name and reuse it instead of duplicating here and L64, L68 for instance? We'd get DRY to improve maintenance and have a nice name that would cause us to not have to think so much when we look at these lines.